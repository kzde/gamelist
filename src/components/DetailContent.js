import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { textSizes, margins } from '../utils/styleUtils';

const Article = styled.article`
  padding: 64px 84px;
  font-size: ${textSizes.normalText};
`;
const Header = styled.header`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;
const Title = styled.span`
  font-size: ${textSizes.titleText};
  font-weight: bold;
`
const Content = styled.div`
  margin: ${margins.large} 0;
  text-align: left;
  line-height: 25px;
`
const DetailContent = (props) => (
  <Article>
    <Header>
      <Title>{props.title}</Title>
      <span>{props.subTitle}</span>
    </Header>
    <Content>{props.content}</Content>
  </Article>
)

DetailContent.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  content: PropTypes.string.isRequired
};
DetailContent.defaultProps = {
  subTitle: ''
};

export default DetailContent;