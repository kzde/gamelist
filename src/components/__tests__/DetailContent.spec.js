import React from 'react';
import renderer from 'react-test-renderer';
import DetailContent from '../DetailContent';

const props = {
  title: 'Title',
  subTitle: 'Subtitle is an option',
  content: 'content here'
};

describe('DetailContent', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(<DetailContent {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})