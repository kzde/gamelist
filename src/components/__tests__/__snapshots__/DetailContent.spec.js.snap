// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailContent should renders correctly 1`] = `
<article
  className="DetailContent__Article-z4i9bw-0 gFleuy"
>
  <header
    className="DetailContent__Header-z4i9bw-1 fscaau"
  >
    <span
      className="DetailContent__Title-z4i9bw-2 isfRBj"
    >
      Title
    </span>
    <span>
      Subtitle is an option
    </span>
  </header>
  <div
    className="DetailContent__Content-z4i9bw-3 bXyuRW"
  >
    content here
  </div>
</article>
`;
